### PR TITLE
le-hy: fix lispy--comint-eval

### PR DESCRIPTION
**Pull request by AlejandroCatalina**
Trying to use lispy to evaluate hy I got with several problems that actually didn't allow me to happily develop hy with lispy in a interactive way.

So I came up with this fix that worked for me. I may have skipped something, in that case, please tell me so I can write the code idiomatically and correct.

Hope this helps!

### CODE REVIEW

**Comment by abo-abo** [comment]
Please test the new change. I tried to make it better. If there's still a problem, just describe what you're trying to eval.

File: le-hy.el [hunk 1]
```diff
@@ -47,39 +47,28 @@
...
-      ;; restore prompt and insert last command
+      (unless (string-equal str "\n=> ")
+        (delete-region command-output-begin (point)))
+      ;; insert last-cmd
+      (insert-string command)
       (goto-char (point-max))
-      (delete-blank-lines)
-      (beginning-of-line)
-      (comint-send-string (get-buffer-process (current-buffer)) "\n")
-      (insert-string last-cmd)
```
Comment (abo-abo) [inline le-hy.el]: insert-string is deprecated

File: le-hy.el [hunk 2]
```diff
@@ -47,39 +47,28 @@
...
       ;; return the shell output
       str)))
 
 (defun lispy--eval-hy (str)
   "Eval STR as Hy code."
   (let ((res (lispy--comint-eval str)))
-    (if (member res '("" "\n"))
+    (if (string= res "\n=> ")
```
Comment (abo-abo) [inline le-hy.el]: The function should not return parts of a prompt. This will make it more useable elsewhere.

**Comment by AlejandroCatalina** [comment]
It doesn't work because it's trying to insert/delete a read only region, namely, the prompt.

Trying to evaluate 
```hy
(import [numpy :as np])
```
whose output is nil.

**Comment by abo-abo** [comment]
> It doesn't work because it's trying to insert/delete a read only region, namely, the prompt.

Works fine for me. Might be an Emacs version issue. What's your version?

**Comment by AlejandroCatalina** [comment]
I'm on 24.5

**Comment by abo-abo** [comment]
Just tested with `emacs-24.5` - everything works fine, with this config: https://github.com/abo-abo/oremacs on Ubuntu 14.04.

**Comment by AlejandroCatalina** [comment]
Thanks! I'll try it here on arch, don't think it's a OS issue, let's see. I'll report when i test it, i image it'll work.

**Comment by AlejandroCatalina** [comment]
It works indeed! I'm diffing what may have caused by failures. Thanks and sorry ;)

**Pull request closed without merging**